Test Python 3.8 in CI

Eric-Arellano opened the pull request:
We claim that Pants works with Python 3.8. We should, therefore, test this.

As part of this change, we remove Ubuntu Trusty and Ubuntu Xenial to only use Ubuntu Bionic, as neither of those come with Python 3.8 by default.

Eric-Arellano committed:
Test Python 3.8 in CI

jsirois commented on .travis.yml:
```diff
@@ -102,20 +105,12 @@ jobs:
       dist: xenial
       env:
         - CACHE_NAME="linux.xenial"
+        - SKIP_PYTHON38_TESTS=true
```
What's the reason for 2 linux shards? OSX makes sense since they ship historically broken interpreters and there is only 1 shipper, Apple. For Linux though there is a large universe of shippers and Ubuntu is a very small slice. Traditionally we have had ~0 problems with linux distro pythons as well.

A further carrot is Bionic already supports 2.7, 3.6, 3.7 & 3.8 - no need for pyenv shenanigans: https://docs.travis-ci.com/user/reference/bionic/#python-support

Eric-Arellano replied:
That's fine with me to drop Xenial, but I'm not sure the motivation. The reason I was keeping it is is that there's little cost for us to support it in CI and it slightly improves our coverage.

For example, in the Venv PR, it looks like Trusty succeeded but for some reason Xenial and Bionic both failed (unclear why). It doesn't slow down our CI much and only adds a couple lines of config to have Xenial.

jsirois replied:
The motivation is to drop pyenv and not pretend we cover linux. Its true 2 is more than 1, but its << 100s.

jsirois replied:
> It doesn't slow down our CI much and only adds a couple lines of config to have Xenial.

Power consumption is a real cost we all regularaly ignore since it feels free. it's not. That's probbly the best argument. This is wasteful for no real benefit.

Eric-Arellano committed:
Drop Xenial

Eric-Arellano committed:
Try fixing macOS shards

Eric-Arellano committed:
Merge branch 'gh-pages' of github.com:pantsbuild/setup into py38

Eric-Arellano committed:
Go back to global env vars for PYENV

I think the PATH wasn't being correctly resolved with the previous setup. While it's not great to set these values for Linux too, it's not the end of the world. Overall, this PR is still an improvement.
